Tests/ArgumentParserUnitTests/HelpGenerationTests+ExpandedSynopses.swift

```diff
@@ -0,0 +1,68 @@
+//===----------------------------------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift Argument Parser open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import ArgumentParserTestHelpers
+@testable import ArgumentParser
+
+extension HelpGenerationTests {
+  enum Fruit: String, ExpressibleByArgument, CaseIterable {
+    case apple, banana, coconut, dragonFruit = "dragon-fruit", elderberry, fig, grape, honeydew
+  }
+  
+  enum Action: String, ExpressibleByArgument, CaseIterable {
+    case purchase, sample, refund = "return"
+  }
+  
+  enum Count: Int, ExpressibleByArgument, CaseIterable {
+    case zero, one, two, three, four, five, six, seven, eight, nine, ten, eleven, twelve, thirteen, fourteen, fifteen, sixteen, seventeen, eighteen, nineteen, twenty
+  }
+  
+  enum Ripeness: String, ExpressibleByArgument, CaseIterable {
+    case under, perfect, over
+  }
+
+  struct FruitStore: ParsableArguments {
+    @Argument(help: "The transaction type")
+    var action: Action = .purchase
+
+    @Argument(help: "The fruit to purchase")
+    var fruit: Fruit
+      
+    @Option(help: "The number of fruit to purchase")
+    var quantity: Count = .one
+
+    @Option(help: "The desired ripeness of fruit")
+    var ripeness: Ripeness = .perfect
+  }
+  
+  func testFruitStoreHelp() {
+    AssertHelp(.default, for: FruitStore.self, equals: """
+            USAGE: fruit_store [<purchase|sample|return>] <fruit> [--quantity <quantity>] [--ripeness <under|perfect|over>]
+
+            ARGUMENTS:
+              <action>                The transaction type (values: purchase, sample,
+                                      return; default: purchase)
+              <fruit>                 The fruit to purchase (values: apple, banana,
+                                      coconut, dragon-fruit, elderberry, fig, grape,
+                                      honeydew)
```

Comment:
It looks like replacing `<action>` with its values makes it harder to see the correspondence between the usage string and the help descriptions. Are there existing tools that include values in the usage string like this?  What would you think of including the argument name – something like:

```
USAGE: fruit_store [<action: purchase|sample|return>] <fruit> ...
```

Reply:
I'm certain that I've used a cli before that had this feature, but over the past two weeks I could not remember what it was.

It did have the issue you describe though, and I like your solution! Another possibility is that ArgumentParser only expand values when showing a short help ([such as gets printed after a `ValidationError`](https://github.com/apple/swift-argument-parser/blob/3dcbdb06ffd57004a37532671eadbcf3aed6d77f/README.md?plain=1#L54-L58)) and leaving the behaviour as-is when printing the complete `--help` output.

Reply:
Ohh the tool I'm thinking of is something I use frequently at work. Instead of an expanded `--help`, it explains multiple subcommands at once and then refers to a `man` page. To paraphrase:

```
> foo bar --help
Usage:
foo alice      --baz <baz_id> [--[no]qux] [--[no-]thbpt] [--fix]
foo bar        --baz <baz_id> [--[no]qux] [--blarpy <blarpy>] [--minimal | --details] [--[no-]box] [--[no-]link] [--eh|--bee|--see] (--regex|--exact|--contains|--suffix) <search_term>
…

foo sub-commands and options are documented in the manual page (`man foo`).
```

I was thinking `ArgumentParser`'s top level help for commands with subcommands is a bit spartan, I wonder if this tool is on to something…